build/local_builder.go

```diff
@@ -313,3 +313,38 @@ func NewLocalBuilder(config *Config) Interface {
 	c.Config = config
 	return c
 }
+
+// used in build stage, where the image still has from layer
+func getBaseLayersFromImage(image v1.Image) (res []string, err error) {
+	if len(image.Spec.Layers) == 0 {
+		return nil, fmt.Errorf("no layer found in image %s", image.Name)
+	}
+
```

Comment:
if layer.0 == common.ImageScratch {
   return ...
  // return fast is better 
}